internal/pkg/instrumentation/manager.go

```diff
@@ -103,8 +103,17 @@ func (m *Manager) FilterUnusedProbes(target *process.TargetDetails) {
 	for name, inst := range m.probes {
 		funcsFound := 0
 		for _, s := range inst.Manifest().Symbols {
-			if _, exists := existingFuncMap[s]; exists {
-				funcsFound++
+			if len(s.DependsOn) != 0 {
+				for _, depends := range s.DependsOn {
+					if _, exists := existingFuncMap[depends]; exists {
+						funcsFound++
+						break
+					}
+				}
+			} else {
+				if _, exists := existingFuncMap[s.Symbol]; exists {
+					funcsFound++
+				}
```

Comment:
This PR https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/654/files is also changing this part. 
I didn't fully understand `DepandsOn` so I'm not sure what is the best approach here.

Reply:
Yeah this section doesn't seem to need to integer count either. @MrAlias can you please merge #654?

Reply:
Since we use bool now, the code is much simpler, we only need to count the probes that don't have dependencies.